### DISCUSSION TITLE
[ZK Stack] -  "Oneshot_runner" task node_storage_initializer failed. Base system contracts mismatched

### DISCUSSION THREAD

**Discussion by Szczepoo13** [zkSync-Community-Hub/zksync-developers#716] [ZK Stack] [answered]
### Module Affected

- [X] System contracts (era-system-contracts)
- [ ] Smart contracts (era-contracts)
- [X] zkEVM (era-zk_evm)
- [ ] Server (zksync-era)
- [ ] Prover (era-boojum)
- [ ] Fee withdrawer (era-fee-withdrawer)
- [ ] Block Explorer (clock-explorer)
- [ ] Portal - Wallet + Bridge (dapp-portal)
- [ ] Other (please specify below)

### Rust Version

rustc 1.83.0-nightly

### Operating System

- [X] Ubuntu
- [ ] MacOS
- [ ] Windows
- [ ] Other (Please specify below)

### Issue Description

Steps I am doing:
zk_inception ecosystem create
zk_inception ecosystem init

When initializing the new ecosystem with the command: **zk_inception ecosystem init** an error pops up:
![error4](https://github.com/user-attachments/assets/dae7505f-3cf4-485b-964b-24eac549559a)
I have done some research and the reason might be that the RocksDB cache is out of sync with Postgres - however, since I am using **zk_inception stack** there is no volume with RocksDB mounted on my docker so I have no idea how to remove the old data.

I am using package version: **zk_inception v0.1.0 (https://github.com/matter-labs/zksync-era/#2a7e72b0)**

### Expected Behavior

The system initializes correctly and I can run a server.

### Current Behavior

When initializing the ecosystem with the command: **zk_inception ecosystem init**
Genesis on the database cannot be performed and an error pops up:

│      Error: One or more tasks failed: [TaskFailed(TaskId("oneshot_runner"), Oneshot task node_storage_initializer failed
│
│      Caused by:
│      Base system contracts mismatched: From Config BaseSystemContractsHashes { bootloader:
│      0x010008e742608b21bf7eb23c1a9d0602047e3618b464c9b59c0fba3b3d7ab66e, default_aa

│      0x01000563374c277a2c1e34659a2a1e87371bb6d852ce142022d497bfb50b9e32 }, Calculated : BaseSystemContractsHashes {
│      bootloader: 0x010008e78b1fb7b6e4423ee126457fdc10f0b77c30311d82569650ed12d79bdc, default_aa:
│      0x0100056372563887f491beb127a25631a4e01557f5edfb2536901225e41e4e7e })]

Screenshot: 
![error4](https://github.com/user-attachments/assets/6e9b5d77-0728-438f-8b43-ae189ba1f2f8)

### Additional Details

I would be grateful for any help! :)

### Prior Research

- [X] Yes, I searched for similar issues and did not find any matching ones.

**Comment by alhenaai** [comment]
It seems like you are encountering a mismatch between the base system contracts specified in your configuration and those calculated during the initialization of your ZKsync ecosystem. This issue typically arises when there is a discrepancy between the expected contract hashes in your configuration files and the actual hashes of the deployed contracts.

Here are a few steps you can take to resolve this issue:

1. **Verify Contract Hashes**: Ensure that the hashes for the bootloader and default account abstraction (AA) contracts in your `configs/contracts.yaml` file match those expected by the system. You might need to update these hashes if they were recently changed in the system contracts repository.

2. **Redeploy Contracts**: If the hashes do not match because the contracts have been updated or incorrectly set, consider redeploying your system contracts. Make sure to update your configuration files with the new hashes.

3. **Clear Docker Volumes**: Since you mentioned there is no volume with RocksDB mounted, ensure that all related Docker volumes are cleared to remove any stale data that might be causing issues. You can do this by running:
   ```bash
   docker volume prune
   ```
   This command will remove all unused Docker volumes. Be cautious, as this will remove all volumes not used by at least one container.

4. **Reinitialize the Ecosystem**: After making the necessary corrections and clearing the volumes, try reinitializing your ecosystem:
   ```bash
   zk_inception ecosystem init --dev
   ```

5. **Check for Updates**: Ensure that your `zk_inception` tool and all related components are up to date. Sometimes, bugs fixed in newer versions might be causing your issue.

If these steps do not resolve the issue, it might be helpful to consult the documentation or seek further assistance from the ZKsync developer community or support channels.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Reply by iscxyz** [reply]
CC @deniallugo

**Reply by alhenaai** [reply]
Thank you for your question! I’m unable to provide an answer at this moment, but I have forwarded your inquiry to our team. Someone will be with you shortly to assist further. Your patience is appreciated!

**Comment by iscxyz** [comment, marked as answer]
Hey @Szczepoo13 - thanks for your question. Based on the information you provided it looks like you might have modified / out of date contracts. Can you please make sure that:
* there are no changes to the contracts
* you are on the latest main
* you've done `git submodule update --recursive` 
* you've run `zkt` before using `zk_toolbox`
* you are starting from clean state - `zks clean all`
